dynamiqs/solvers/solver.py

```diff
@@ -41,45 +43,48 @@ def __init__(
         self.rdtype = self.options.rdtype
         self.device = self.options.device
 
-        # initialize saving logic
-        self._init_time_logic()
+        # initialize time logic
+        self.tstop = torch.cat((self.tsave, self.tmeas)).unique().sort()[0]
```

Comment:
```suggestion
        self.tstop = torch.cat((self.tsave, self.tmeas)).unique()
```
FYI you don't need this if `tsave` and `tmeas` are already sorted. But I forget if we already check this, and "au pire ça mange pas de pain"

Reply:
No that's a good comment, we do check if they are sorted in `sesolve`, `mesolve`.

Reply:
But `unique` sorts everything no matter what.